cmd/create/accountroles/creators.go

```diff
@@ -118,6 +118,13 @@ type unmanagedPoliciesCreator struct{}
 
 func (up *unmanagedPoliciesCreator) createRoles(r *rosa.Runtime, input *accountRolesCreationInput) error {
 	for file, role := range aws.AccountRoles {
+		// If hosted-cp flag was passed, create only the hosted CP installer policy
+		if r.HostedCP != nil {
+			if *r.HostedCP && file == aws.InstallerAccountRole {
```

Comment:
Here we skip also the role creation in line `137` **r.AWSClient.EnsureRole**.
We have another function to create roles with managed policies https://github.com/openshift/rosa/blob/master/cmd/create/accountroles/creators.go#L54.